install.ts

```diff
@@ -0,0 +1,205 @@
+  const url: string = tag ? `${TAG_RELEASE_URL}/${tag}` : LATEST_RELEASE_URL;
+  let filename: string;
+  switch (platform.os) {
+    case "linux":
+      filename = LINUX_GZIP;
+      break;
+    case "mac":
+      filename = OSX_GZIP;
+      break;
+    case "win":
+      filename = WIN_ZIP;
+      break;
+    default:
+      throw Error(`Unsupported operating system ${platform.os}`);
+  }
+  const res: any = await follow(url); // TODO: annotate deno Response
+  const link: string = (await res.text())
+    .split(/\r?\n/)
+    .find((line: string) => line.includes("href") && line.includes(filename));
+  if (!link) panic(Error(`Unable to find ${filename} @ ${url}`));
+  const match: string = link.replace(/^.*href=(?:"|')([^"']*)(?:"|').*$/, "$1");
+  if (!/^\/denoland/.test(match))
+    panic(Error(`Unable to find ${filename} @ ${url}`));
+  return {
+    url: `https://github.com${match}`,
+    tag: match.replace(/^.*(v\d+\.\d+\.\d+).*$/, "$1")
+  };
+}
+
+async function tempDownload(
+  tempDir: string,
+  url: string,
+  suffix: string
+): Promise<string> {
+  const res: any = await follow(url); // TODO: annotate deno Response
+  const tempFile: string = join(tempDir, `${Date.now()}.${suffix}`);
+  await writeFile(tempFile, new Uint8Array(await res.arrayBuffer()));
+  return tempFile;
+}
+
+async function unpackBin(archive: string): Promise<void> {
+  await mkdirp(DENO_BIN_DIR);
+  let args: string[];
+  if (WIN32) {
+    await rename(DENO_BIN, OLD_DENO_BIN);
+    args = [
+      "powershell.exe",
+      "-Command",
+      `Expand-Archive "${archive}" -DestinationPath "${DENO_BIN_DIR}"`
+    ];
+  } else {
+    args = ["gunzip", "-d", archive];
+  }
+  const child: Process = run({ args });
+  const childStatus: ProcessStatus = await child.status();
+  if (!childStatus.success)
+    panic(Error(`(g)unzip failed. ${args} -> ${childStatus.code}`));
+  child.close();
+  if (!WIN32) {
+    if (platform.os === "linux") await rename(DENO_BIN, OLD_DENO_BIN);
+    await copyFile(archive.replace(/\.gz$/, ""), DENO_BIN);
+  }
+}
+
+async function makeHandy(): Promise<void> {
+  if (WIN32) {
+    if (!procEnv.Path.toLocaleLowerCase().includes(DENO_BIN_DIR)) {
+      let updatedPath: string = procEnv.Path;
+      if (procEnv.Path.endsWith(";")) updatedPath += DENO_BIN_DIR;
+      else updatedPath += `;${DENO_BIN_DIR}`;
+      const args: string[] = [
+        "powershell.exe",
+        "-Command",
+        `[Environment]::SetEnvironmentVariable("PATH","${updatedPath}",` +
+          `[EnvironmentVariableTarget]::User)`
+      ];
+      const ps: Process = run({ args });
+      const psStatus: ProcessStatus = await ps.status();
+      if (!psStatus.success)
+        panic(Error(`Unable to edit PATH. ${args} -> ${psStatus.code}`));
+      ps.close();
+    }
+  } else {
+    await chmod(DENO_DIR, 0o744);
+    await chmod(DENO_BIN, 0o744);
+    if (!procEnv.Path.toLocaleLowerCase().includes(DENO_BIN_DIR)) {
```

Comment:
This else-block is for non-win32 systems. And I don't think file paths are case-insensitive there, so I think the toLocaleLowerCase needs to be removed here.